lib/pushmi_pullyu/aip_downloader.rb

```diff
@@ -0,0 +1,147 @@
+require 'pushmi_pullyu/aip_downloader_files_directories'
+require 'pushmi_pullyu/aip_logger'
+require 'pushmi_pullyu/fedora_object_fetcher'
+require 'pushmi_pullyu/solr_fetcher'
+
+# Download all of the metadata/datastreams and associated data
+# related to an object
+
+class PushmiPullyu::AipDownloader
+
+  include PushmiPullyu::AipDownloaderFilesDirectories
+
+  attr_reader :noid, :config, :logger, :aip_logger, :fetcher,
+              :aipcreation_log
+
+  attr_accessor :got_characterization, :got_thumbnail,
+                :got_fedora3foxml, :got_fedora3foxml_metadata
```

Comment:
The ability to fetch these is optional. If we can fetch them, we will archive them, if not we don't raise an error. These track which ones of the optional pieces were successfully fetched. That said, they aren't used anywhere else in the code, so may not be useful.